tests/test_etherscan.py

```diff
@@ -36,27 +39,38 @@ def get(base_uri, params=None, *args, **kwargs):
     return get_patch
 
 
-def test_get_address_url(explorer):
-    expected = f"https://etherscan.io/address/{ADDRESS}"
-    actual = explorer.get_address_url(ADDRESS)
+@pytest.mark.parametrize(
+    "network,expected_prefix,address",
+    [(NETWORKS[0], "etherscan.io", ADDRESS), (NETWORKS[1], "ropsten.etherscan.io", ADDRESS)],
+)
+def test_get_address_url(network, expected_prefix, address):
+    expected = f"https://{expected_prefix}/address/{ADDRESS}"
+    explorer = get_explorer(network)
+    actual = explorer.get_address_url(address)  # type: ignore
     assert actual == expected
```

Comment:
hmm, maybe a better test would be to see if it gives you a 404

Reply:
Hmm, I disagree. Making real network requests in tests is not a good thing.

Reply:
Unless you control the service. Making requests to Etherscan / Infura, etc. in tests, even integration, is not really necessary and just complicates stuff. If Etherscan or Infura break, that is not on us. That should not cause test failures in `ape`. Relying on services in tests can lead to other behaviors too. It is out of scope!

What would be _in scope_ though, is testing how we handle 404s or something (we are not doing that here but `ape` maybe we would?)

Reply:
fair